packages/venia-ui/lib/components/Navigation/navHeader.js

```diff
@@ -1,38 +1,54 @@
-import React, { Component, Fragment } from 'react';
-import { func, shape, string } from 'prop-types';
+import React, { Fragment } from 'react';
+import { bool, func, shape, string } from 'prop-types';
+import {
+    ArrowLeft as ArrowLeftIcon,
+    Menu as MenuIcon,
+    X as CloseIcon
+} from 'react-feather';
 
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import Icon from '../Icon';
-import { ArrowLeft as ArrowLeftIcon, X as CloseIcon } from 'react-feather';
 import Trigger from '../Trigger';
 import defaultClasses from './navHeader.css';
 
-class NavHeader extends Component {
-    static propTypes = {
-        classes: shape({
-            title: string
-        }),
-        onBack: func.isRequired,
-        onClose: func.isRequired
-    };
+const titles = {
+    CREATE_ACCOUNT: 'Create Account',
+    FORGOT_PASSWORD: 'Forgot Password',
+    MY_ACCOUNT: 'My Account',
+    SIGN_IN: 'Sign In',
+    MENU: 'Main Menu'
+};
 
-    render() {
-        const { classes, onBack, onClose, title } = this.props;
+const NavHeader = props => {
+    const { isTopLevel, onBack, onClose, view } = props;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-        return (
-            <Fragment>
-                <Trigger key="backButton" action={onBack}>
-                    <Icon src={ArrowLeftIcon} />
-                </Trigger>
-                <h2 key="title" className={classes.title}>
-                    <span>{title}</span>
-                </h2>
-                <Trigger key="closeButton" action={onClose}>
-                    <Icon src={CloseIcon} />
-                </Trigger>
-            </Fragment>
-        );
-    }
-}
+    const title = titles[view] || titles.MENU;
+    const backIcon = isTopLevel && view === 'MENU' ? MenuIcon : ArrowLeftIcon;
```

Comment:
This is a tricky problem.

- If we have a back arrow at the root, then while you're clicking back from nested categories, you're bound to click it and close the drawer inadvertently before you realize you're at the root.
- If we change it to a hamburger icon, it becomes clear that you're reached the root and the only thing it could reasonably do is close the drawer. Least surprise, for sure, but redundant with the close icon.
- If we remove the button and preserve the space, the blank space looks wrong.
- If we remove the button and don't preserve the space, the title moves as you navigate.

Other options:

- When the drawer is open and at the root, change the back arrow to a close icon and hide the righthand close icon.
- When the drawer is open and at the root, show a small, noninteractive Venia logo in place of the back arrow.

Reply:
I think we need @soumya-ashok's input but she is traveling. I say leave it as is for now and open an issue with `needs-ux`.